nautilus_core/network/src/websocket.rs

```diff
@@ -86,7 +86,11 @@ impl WebSocketClientInner {
     /// Connects with the server creating a tokio-tungstenite websocket stream.
     #[inline]
     pub async fn connect_with_server(url: &str) -> Result<(MessageWriter, MessageReader), Error> {
```

Comment:
You want to pass a clone of the stored `Vec<(String, String)>` here. Then consume and add the key value pairs to the request.